src/MinimalDelegation.sol

```diff
@@ -142,9 +143,11 @@ contract MinimalDelegation is IERC7821, ERC1271, EIP712, ERC4337Account, Receive
         _payEntryPoint(missingAccountFunds);
         (bytes32 keyHash, bytes calldata signature) = userOp.signature.unwrap();
 
-        IHook hook = keySettings[keyHash].hook();
+        Settings settings = keySettings[keyHash];
+        // TODO: should this be return 0 or revert
+        if (settings.isExpired()) revert IKeyManagement.KeyExpired();
```

Comment:
I think revert is good

Its kinda funny that the settings can expire? feels like odd that its not really clear its the KEY that has expired?